packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -520,14 +520,16 @@ class CircularProgressIndicator extends ProgressIndicator {
   @override
   Color? get backgroundColor => super.backgroundColor;
 
+  /// {@template flutter.material.CircularProgressIndicator.strokeWidth}
   /// The width of the line used to draw the circle.
+  /// {@endtemplate}
   final double strokeWidth;
 
   @override
-  State<CircularProgressIndicator> createState() => _CircularProgressIndicatorState();
+  State<CircularProgressIndicator> createState() => _CircularProgressIndicatorState<CircularProgressIndicator>();
 }
 
-class _CircularProgressIndicatorState extends State<CircularProgressIndicator> with SingleTickerProviderStateMixin {
+class _CircularProgressIndicatorState<T extends CircularProgressIndicator> extends State<T> with SingleTickerProviderStateMixin {
```

Comment:
Is there a reason this needs to be made a type parameter? `RefreshProgressIndicator` is already a subtype of `CircularProgressIndicator`.

Reply:
Well I have added a new property on the subclass without this I can't access to it. I could cast the super class as the subclass, but I think this is can be useful for the future.